triton-client/src/components/DatasetCard.tsx

```diff
@@ -27,7 +27,9 @@ function DatasetCard({ datasetID }: DatasetCardProps) {
 	const constants = useAppSelector(selectConstants)
 	const readsetsByDatasetId = useAppSelector((state) => state.readsetsState.readsetsByDatasetId)
 	const readsetsById = useAppSelector((state) => state.readsetsState.readsetsById)
-  const [activeRequest, setActiveRequest] = useState<DownloadRequest>()
+  const activeRequest = useMemo<DownloadRequest | undefined>(() => {
+    return dataset?.requests[0] ?? undefined
```

Comment:
Shouldn't you check for the length of the requests first?

Reply:
the length was just to know if it was defined ... if it is not defined i will get undefined.

Reply:
That's not how array works in javascript. If the array is empty, you will most definitely get index out of bound.